pages/partials/questionFooter.ejs

```diff
@@ -115,3 +115,31 @@
         $(this).addClass('disabled');
     });
 </script>
+<script>
+    const form = $('form.question-form');
+    const getForm = () => form.serialize();
+
+    // Wait for DOM (ace, file upload, etc.)
+    $(document).ready(() => {
+        // Set form state on load and submit
+        var initialForm = getForm();
+        form.submit(() => {
+            initialForm = getForm();
+        });
+
+        // Check form state on unload
+        $(window).bind('beforeunload', (event) => {
+            const currentForm = getForm();
+            if(initialForm != currentForm) {
+                // MDN recommendation (not fully supported)
+                (event || window.event).preventDefault();
+
+                // Chrome legacy
+                (event || window.event).returnValue = '';
+
+                // Fallback
+                return '';
```

Comment:
New:
* added `preventDefault()` (per MDN)
* added comments to explain the fallback chain
* changed the return values from `null` to `''` (as used by MDN, though seems to work either way)